pkg/ipam-node/handlers/allocate.go

```diff
@@ -219,6 +216,48 @@ func (h *Handlers) allocateInPool(poolName string, reqLog logr.Logger,
 	}, nil
 }
 
+func newRangeSetAndDependenciesForPoolCfg(poolCfg *pool.Pool) (*allocator.RangeSet, net.IP, *net.IPNet, net.IP, error) {
+	rangeStart := net.ParseIP(poolCfg.StartIP)
+	if rangeStart == nil {
+		return nil, nil, nil, nil, errors.New("invalid rangeStart")
+	}
+	rangeEnd := net.ParseIP(poolCfg.EndIP)
+	if rangeEnd == nil {
+		return nil, nil, nil, nil, errors.New("invalid rangeEnd")
+	}
+	_, subnet, err := net.ParseCIDR(poolCfg.Subnet)
+	if err != nil || subnet == nil || subnet.IP == nil || subnet.Mask == nil {
+		return nil, nil, nil, nil, errors.New("invalid subnet")
+	}
+	gateway := net.ParseIP(poolCfg.Gateway)
+	rangeSet := &allocator.RangeSet{allocator.Range{
+		RangeStart: rangeStart,
+		RangeEnd:   rangeEnd,
+		Subnet:     cniTypes.IPNet(*subnet),
+		Gateway:    gateway,
+	}}
+	return rangeSet, rangeStart, subnet, gateway, nil
+}
+
+func setAllocMetadata(allocMeta *types.ReservationMetadata, paramsMetadata *nodev1.IPAMMetadata) {
+	if paramsMetadata != nil {
+		allocMeta.PodUUID = paramsMetadata.K8SPodUid
+		allocMeta.PodName = paramsMetadata.K8SPodName
+		allocMeta.PodNamespace = paramsMetadata.K8SPodNamespace
+		allocMeta.DeviceID = paramsMetadata.DeviceId
+	}
+}
+
+// IPOffsetBy returns the IP that is offseted by index from the given IP
+func IPOffsetBy(ip net.IP, index int32) net.IP {
```

Comment:
Let's use this one https://github.com/Mellanox/nvidia-k8s-ipam/blob/6649bfc810caa75889b7c4592cd90e67f64aaa78/pkg/ip/cidr.go#L35